Reorganise wallet setup screens

mankyKitty opened the pull request:
As part of the redesign process, the workflow for the wallet creation / recovery has been updated to have the passwords fields on the same screen as the recovery phrase creation/input.

The workflow has been updated to suit and a small timer has been added on the password validation steps that is gated on both fields be non-empty on a 0.4 second delay. This prevents the validation from spamming the user and also means they don't have to wait until they've clicked next before they get some feedback about the password input.

Removed the `h1` from the password fields but there are no other styling changes so it looks pretty janky at the moment.

Commit:
Reorganise wallet setup/recovery screens to new ordering

alexfmpe commented on desktop/src/Desktop/Setup.hs:
```diff
@@ -176,36 +192,46 @@ confirmPhrase mnemonic = Workflow $ do
-    ]
+
+  pure
+    ( fmapMaybe id $ current dPassword <@ gate (current done) next <> skip
```
`tagMaybe (current dPassword) $ gate (current done) next <> skip`

alexfmpe commented on desktop/src/Desktop/Setup.hs:
```diff
@@ -176,36 +192,46 @@ confirmPhrase mnemonic = Workflow $ do
-    , previous <$ back
-    )
+
+    let (err, pass) = fanEither $ attachWith (\p _ -> uncurry checkPassword p) (liftA2 (,) p1 p2) eCheckPassword
```
`let (err, pass) = fanEither $ checkPassword p <$> p1 <*> p2 <@ eCheckPassword`

alexfmpe commented on desktop/src/Desktop/Setup.hs:
```diff
@@ -133,21 +142,28 @@ createNewWallet mMnemonic = Workflow $ do
+  ePassword <- setPassword (fmap (fmap (\mn -> Crypto.sentenceToSeed mn Crypto.english "") . hush) mnemonic)
+    >>= holdDyn Nothing . fmap pure
+
+  let nextGate = liftA2 (||) (fmap isNothing dPassword) (fmap not stored)
```
`gate beh ev` allows `ev` through when `beh` is `True`, so I thought `nextGate` would be `True` when allowing the "next" action. Maybe call it `disabledNext`

Commit:
Added 12 word passphrase widget for setup/recovery steps

Commit:
PR feedback update

Commit:
Fix bug on confirm phrase screen.

Dedupe some code into helper functions.

Remove redundant dynamic that was causing the password to fail validation.

Update the confirm phrase screen to use the new widget and validate it against the
generated sentence.

Commit:
Change wallet passphrase hiding behaviour.

This behaviour now comes from css, with some classes swapped out to avoid the hiding
behaviour on the wallet recovery screen.

Commit:
Simplified the passphrase widget.

Removed the use of the password type and cleaned up the css classes and removed the need
for the recursive section.

Commit:
Reworked wallet setup recover flow.

Leverage switching to not have to juggle the `Maybe Mnemonicsentence` values and create
opportunities for nicer error handling.

Commit:
Clean up workflow more and prepare for restyle

Commit:
Initial styling, needs correct colours

Also needs to work out how to create the staggered menu at the top of the page.

Commit:
Lots of styling and wallet create flow changes.

Applying new styling based on the redesigns provided by Kadena. Added additional screens
and messages. Added some smoothing effects to the password error messages so it's less
jarring when they appear but still noticeable.

Commit:
Export copyToClipboard function.

Implement copy to clipboard behaviour on the create wallet workflow.

Commit:
Add additional wallet background

Commit:
Styling adjustments for splashscreen

Add new blue kadena logo.
Add correct font for 'Chainweaver' on splash screen.
Add text for 'Chainweaver By Kadena' on splash screen.

Commit:
More styles applied to wallet creation.

Added fonts for 'Chainweaver by Kadena'.

Moved new images into position.

Rebuilt the page to use a CSS grid layout for easier placement of the header and back
button.

Commit:
Started restyle of the wallet recover screen

Commit:
Added padlock icons to password inputs

Fixed up some warnings for redundant imports in `desktop/Setup.hs`.

Changed the colour of the checkboxes to match the default Kadena blue.

removed some dev hacks for auto-filling fields.

Commit:
Reset minimum password length

Commit:
Fix up identing, remove a commented line

tomsmalley reviewed (approved):
This looks solid. There are a couple of things that need altering, but I'll create new tickets for that.

Merging into a new `redesign` branch.

eskimor commented on backend/sass/common.blocks/_wallet.scss:
```diff
@@ -4,6 +4,166 @@
+  grid-template-columns: 1fr 1fr 1fr 1fr;
+  grid-template-rows: 0.2fr 1.6fr;
+  grid-template-areas: "back header header ." ". wrapper wrapper .";
+}
```
Sorry, just stumbled upon this. This should really not be here. Either that fullscreen is an element of the wallet block, then it should be named wallet__fullscreen or it is a block of its own. In that case it should have its own file _fullscreen.

tomsmalley replied:
Ah, I totally missed the file name. The setup screen code should probably have a `_setup` file, since it's the only place we have the fullscreen stuff anyway. I'll add a ticket to move it

eskimor replied:
Sounds good! Thanks! :-)
